samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/DocumentImportController.cs

```diff
@@ -76,108 +76,182 @@ private enum SupportedFileType
         [FromServices] IHubContext<MessageRelayHub> messageRelayHubContext,
         [FromForm] DocumentImportForm documentImportForm)
     {
-        var formFile = documentImportForm.FormFile;
-        if (formFile == null)
+        try
         {
-            return this.BadRequest("No file was uploaded.");
+            await this.ValidateDocumentImportFormAsync(documentImportForm);
         }
+        catch (ArgumentException ex)
+        {
+            return this.BadRequest(ex.Message);
+        }
+
+        this._logger.LogInformation("Importing {0} document(s)...", documentImportForm.FormFiles.Count);
 
-        if (formFile.Length == 0)
+        // TODO: Perform the import in parallel.
+        DocumentMessageContent documentMessageContent = new();
+        foreach (var formFile in documentImportForm.FormFiles)
         {
-            return this.BadRequest("File is empty.");
+            var success = await this.ImportDocumentHelperAsync(kernel, formFile, documentImportForm);
+            documentMessageContent.AddDocument(
+                formFile.FileName,
+                this.GetReadableByteString(formFile.Length),
+                success);
         }
 
-        if (formFile.Length > this._options.FileSizeLimit)
+        // Broadcast the document uploaded event to other users.
+        if (documentImportForm.DocumentScope == DocumentImportForm.DocumentScopes.Chat)
         {
-            return this.BadRequest("File size exceeds the limit.");
+            var chatMessage = await this.TryCreateDocumentUploadMessage(
+                documentMessageContent,
+                documentImportForm);
+            if (chatMessage == null)
+            {
+                // It's Ok to have the message not created.
+                return this.Ok();
+            }
+
+            var chatId = documentImportForm.ChatId.ToString();
+            await messageRelayHubContext.Clients.Group(chatId)
+                .SendAsync(ChatDocumentUploadedClientCall, chatMessage, chatId);
+
+            return this.Ok(chatMessage);
         }
 
+        await messageRelayHubContext.Clients.All.SendAsync(
+            GlobalDocumentUploadedClientCall,
+            documentMessageContent.ToFormattedStringNamesOnly(),
+            documentImportForm.UserName
+        );
+
+        return this.Ok();
+    }
+
+    #region Private
+
+    /// <summary>
+    /// Validates the document import form.
+    /// </summary>
+    /// <param name="documentImportForm">The document import form.</param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentException">Throws ArgumentException if validation fails.</exception>
+    private async Task ValidateDocumentImportFormAsync(DocumentImportForm documentImportForm)
+    {
+        var formFiles = documentImportForm.FormFiles;
+
+        if (formFiles.Count == 0)
+        {
+            throw new ArgumentException("No file was uploaded.");
+        }
+        else if (formFiles.Count > this._options.FileCountLimit)
```

Comment:
What is the purpose behind the max file count? Does it stall the app or something?

Reply:
It's going to be extremely slow because we are uploading the documents in series. Setting a cap here to have at least some restriction.